plugins/junit/src/com/intellij/execution/junit/codeInsight/JUnit5MalformedOrderedTestInspection.kt

```diff
@@ -0,0 +1,165 @@
+          return
+        }
+
+        val testMethodOrderAnnotation = detectProblemPsiClassWithoutTestMethodOrderAnnotation(
+          psiClass,
+          method.nameIdentifier ?: return,
+          "Test method with @Order inside class without @TestMethodOrder") ?: return
+
+        detectProblemWithTestMethodOrderAnnotationValue(
+          testMethodOrderAnnotation,
+          psiClass,
+          method.nameIdentifier ?: return,
+          "Test method with @Order inside class with @TestMethodOrder but without @Order support")
+      }
+
+      /**
+       * @return `null` when problem has been found, otherwise no problem decided returning instance of [@PsiAnnotation]
+       * on which decision could be made
+       */
+      private fun detectProblemPsiClassWithoutTestMethodOrderAnnotation(
+        classWithTestMethodOrderer: PsiClass,
+        problemOnPsiElement: PsiElement,
+        problemDescription: String
+      ): PsiAnnotation? {
+        val testMethodOrderAnnotation = AnnotationUtil.findAnnotationInHierarchy(classWithTestMethodOrderer, setOf(
+          JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST_METHOD_ORDER))
+        if (testMethodOrderAnnotation == null) {
+          holder.registerProblem(
+            problemOnPsiElement,
+            problemDescription,
+            object : AddAnnotationFix(
+              JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST_METHOD_ORDER,
+              classWithTestMethodOrderer,
+              createAnnotationValues(classWithTestMethodOrderer)) {
+              override fun getFamilyName(): String = "Add @TestMethodOrder with @Order support"
+            })
+          return null
+        }
+        else {
+          return testMethodOrderAnnotation
+        }
+      }
+
+      private fun detectProblemWithTestMethodOrderAnnotationValue(
+        testMethodOrderAnnotation: PsiAnnotation,
+        placeOnPsiClass: PsiClass,
+        problemOnPsiElement: PsiElement,
+        problemDescription: String
+      ) {
+        val value = testMethodOrderAnnotation.findAttributeValue("value")
+                      as PsiClassObjectAccessExpression? ?: return
+
+        val valuePsiClass = PsiTypesUtil.getPsiClass(value.operand.type) ?: return
+        if (PsiClassImplUtil.isClassEquivalentTo(valuePsiClass, psiClassOrderAnnotationOrderer)) {
+          // annotation value is known correct MethodOrderer implementation
+          return
+        }
+
+        val packageName = PsiUtil.getPackageName(valuePsiClass) ?: return
+        if (packageName.startsWith("org.junit.jupiter.api")) {
+          // other JUnit5's MethodOrderer implementations ignore @Order annotation
+          holder.registerProblem(
+            problemOnPsiElement,
+            problemDescription,
+            object : RemoveAnnotationQuickFix(testMethodOrderAnnotation, placeOnPsiClass) {
+              // 1st LocalQuickFix's familyName is used to show to the user
```

Comment:
I am also not sure that when I want to use "composite" quick-fixes with one element in UI that the correct thing is to put familyName on first element. This seems to my that I am doing something wrong.

Reply:
i also tried 
```
          holder.registerProblem(
            problemOnPsiElement,
            problemDescription,
            object : LocalQuickFix {
              override fun getFamilyName(): String = "Replace @TestMethodOrder value to support @Order"

              override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
                RemoveAnnotationQuickFix(testMethodOrderAnnotation, placeOnPsiClass).applyFix(project, descriptor)
                AddAnnotationFix(
                  JUnitCommonClassNames.ORG_JUNIT_JUPITER_API_TEST_METHOD_ORDER,
                  placeOnPsiClass,
                  createAnnotationValues(placeOnPsiClass)).applyFix(project, descriptor)
              }
            })
```
but without luck (with `PsiInvalidElementAccessException`)